### ISSUE TITLE
prediction failure on Cityscapes images

### ISSUE THREAD

**Issue by remtav** [nyoki-mtl/pytorch-segmentation#10]
While testing the code with supplied Cityscapes config file and pretrained weights, about 1 prediction out of 80 fails completely. See attached pictures. Did not occur on original tensorflow model.

![191_img_2019-02-28](https://user-images.githubusercontent.com/34774759/53584844-91bdc180-3b52-11e9-98a9-fbeff777ef9d.jpg)
![191_pred_2019-02-28_color](https://user-images.githubusercontent.com/34774759/53584598-f6c4e780-3b51-11e9-9630-6a83b59687aa.png)

![212_img_2019-02-28](https://user-images.githubusercontent.com/34774759/53584841-91252b00-3b52-11e9-99a3-c6d97ace8cfc.jpg)
![212_pred_2019-02-28_color](https://user-images.githubusercontent.com/34774759/53584599-f6c4e780-3b51-11e9-9de6-d2e8c4168976.png)

![017_img_2019-02-28](https://user-images.githubusercontent.com/34774759/53584842-91252b00-3b52-11e9-8bdc-79f1a5b6ce79.jpg)
![017_pred_2019-02-28_color](https://user-images.githubusercontent.com/34774759/53584600-f75d7e00-3b51-11e9-91c0-20b4c0a3be67.png)

![097_img_2019-02-28](https://user-images.githubusercontent.com/34774759/53584843-91252b00-3b52-11e9-9d6c-88186ea54ea2.jpg)
![097_pred_2019-02-28_color](https://user-images.githubusercontent.com/34774759/53584601-f75d7e00-3b51-11e9-89f1-5b15e9e57c3d.png)

**Comment by zzllc3** [comment]
I met the same problem on the other dataset. Do you have any idea to solve this problem?

**Comment by sunlibocs** [comment]
I also meet this problem. Has anyone found a solution?

**Comment by nyoki-mtl** [comment]
Sorry for late reply.
This problem is because the default value of tensorflow batch normalization eps is different.
So, it can be solved by calling model.update_bn_eps ().
https://github.com/nyoki-mtl/pytorch-segmentation/blob/7a30cabb1e169f819329a5e3bd9d39f21d41c1c1/src/eval_cityscapes.py#L60

**Comment by PierfrancescoArdino** [comment]
Sorry @nyoki-mtl when you have to call model.update_bn_eps() in the evaluation/prediction?

**Comment by nyoki-mtl** [comment]
@PierfrancescoArdino 
If you want to use the trained model converted from tensorflow, you should call update_bn_eps.
But, when you train or predict on your model, you don't need to do.

**Comment by PierfrancescoArdino** [comment]
I've tried only the "eval_cityscapes.py" script without changing anything with the converted trained model from TF and I have the same problems of the first post.
